src/main/java/org/opensearch/eval/engine/OpenSearchEngine.java

```diff
@@ -218,6 +227,94 @@ public Double getJudgmentValue(final String judgmentsId, final String userQuery,
+
+        final MatchAllQuery matchAllQuery = new MatchAllQuery.Builder().build();
+
+        final FunctionScoreQuery functionScoreQuery = new FunctionScoreQuery.Builder()
+                .query(matchAllQuery.toQuery())
+                .functions(List.of(functionScore))
+                .build();
+
+        final ExistsQuery existsQuery = new ExistsQuery.Builder()
+                .field(USER_QUERY_FIELD)
+                .build();
+
+        final BoolQuery boolQuery = new BoolQuery.Builder()
+                .must(q -> q.exists(existsQuery))
+                .must(q -> q.functionScore(functionScoreQuery))
+                .build();
+
+        final SearchRequest searchRequest = new SearchRequest.Builder()
+                .index(Constants.UBI_QUERIES_INDEX_NAME)
+                .query(boolQuery.toQuery())
+                .collapse(FieldCollapse.of(c -> c.field(USER_QUERY_FIELD)))
+                .size(n)
+                .build();
+
+        final SearchResponse<UbiQuery> searchResponse = client.search(searchRequest, UbiQuery.class);
+
+        final Map<String, Long> querySet = new HashMap<>();
+
+        // Is having the frequency for the random queries useful?
+        searchResponse.hits().hits().forEach(hit -> {
+            LOGGER.info("Adding random user query: {}", hit.source().getUserQuery());
+            querySet.put(hit.source().getUserQuery(), 1L);
+        });
+
+        return querySet;
+
+    }
+
+    @Override
+    public Map<String, Long> getUbiQueries(final int n) throws IOException {
+
+        final Map<String, Long> querySet = new HashMap<>();
+
+        final Aggregation userQueryAggregation = Aggregation.of(a -> a
+                .terms(t -> t
+                        .field(USER_QUERY_FIELD)
+                        .size(n)
+                )
+        );
+
+        final Map<String, Aggregation> aggregations = new HashMap<>();
+        aggregations.put("By_User_Query", userQueryAggregation);
+
+        final ExistsQuery existsQuery = new ExistsQuery.Builder()
+                .field(USER_QUERY_FIELD)
+                .build();
+
+        final SearchRequest searchRequest = new SearchRequest.Builder()
+                .index(Constants.UBI_QUERIES_INDEX_NAME)
+                .query(existsQuery.toQuery())
+                .aggregations(aggregations)
+                .from(0)
+                .size(0)
+                .build();
+
+        final SearchResponse<Void> searchResponse = client.search(searchRequest, Void.class);
+
+        final Map<String, Aggregate> aggs = searchResponse.aggregations();
+        final StringTermsAggregate byUserQuery = aggs.get("By_User_Query").sterms();
+        final List<StringTermsBucket> byActionBuckets = byUserQuery.buckets().array();
+
+        for (final StringTermsBucket bucket : byActionBuckets) {
+
+            LOGGER.info("Adding user query to query set: {} with frequency {}", bucket.key(), bucket.docCount());
+            querySet.put(bucket.key(), bucket.docCount());
```

Comment:
need a check for empty queries, otherwise fails when encountering an empty string as a query with the mapper parsing exception.